packages/fuels-contract/src/script.rs

```diff
@@ -259,6 +178,84 @@ impl Script {
         instructions.iter().copied().collect::<Vec<u8>>()
     }
 
+    /// Returns the assets and contracts that will be consumed (inputs) and created (outputs)
+    /// by the transaction
+    async fn get_transaction_inputs_outputs(
+        calls: Vec<&ContractCall>,
+        wallet: &LocalWallet,
+    ) -> (Vec<Input>, Vec<Output>) {
+        let mut inputs: Vec<Input> = vec![];
+        let mut outputs: Vec<Output> = vec![];
+
+        // Get all unique contract ids
+        let mut contract_ids = HashSet::new();
+        calls.iter().for_each(|call| {
```

Comment:
A more functional approach, not sure how readable though:
```rust
  let contract_ids: HashSet<ContractId> = calls
            .iter()
            .flat_map(|call| {
                let mut ids = call
                    .external_contracts
                    .as_ref()
                    .cloned()
                    .unwrap_or_default();
                ids.insert(call.contract_id);
                ids
            })
            .collect();
```

Also you can do without the `clone()` in `call.external_contracts.clone()` by doing `&call.external_contracts` since .extend does its own cloning inside.